store.go

```diff
@@ -3682,3 +3753,30 @@ func (s *store) Free() {
 		}
 	}
 }
+
+func searchLayer(ls drivers.AdditionalLayerStore, d digest.Digest, ref string) (*Layer, string, error) {
+	var elems []string
+	if ls.WithReference {
+		elems = append(elems, base64.StdEncoding.EncodeToString([]byte(ref)))
+	}
+	elems = append(elems, d.Encoded())
```

Comment:
Doesn’t this discard the algorithm ID? Why is that safe?

Reply:
Fixed this to preserve the algorithm ID.